sdk/core/core-http/lib/util/utils.ts

```diff
@@ -246,16 +270,145 @@ export function isDuration(value: string): boolean {
  * @param {string} replaceValue The value to replace searchValue with in the value argument.
  * @returns {string | undefined} The value where each instance of searchValue was replaced with replacedValue.
  */
-export function replaceAll(value: string | undefined, searchValue: string, replaceValue: string): string | undefined {
+export function replaceAll(
+  value: string | undefined,
+  searchValue: string,
+  replaceValue: string
+): string | undefined {
   return !value || !searchValue ? value : value.split(searchValue).join(replaceValue || "");
 }
 
 /**
- * Determines whether the given enity is a basic/primitive type
+ * Determines whether the given entity is a basic/primitive type
  * (string, number, boolean, null, undefined).
- * @param value Any entity
- * @return boolean - true is it is primitive type, false otherwise.
+ * @param {any} value Any entity
+ * @return {boolean} - true is it is primitive type, false otherwise.
  */
 export function isPrimitiveType(value: any): boolean {
   return (typeof value !== "object" && typeof value !== "function") || value === null;
 }
+
+/**
+ * Checks if given `text` starts with the specified `prefix`
+ * @param text Input string
+ * @return {boolean} - true if yes, false otherwise.
+ */
+export function stringStartsWith(text: string, prefix: string) {
+  if (prefix == null) {
+    return true;
+  }
+
+  return text.substr(0, prefix.length) === prefix;
+}
+
+/**
+ * Determines whether the given `value` is a `Date` object or not.
+ * @param {any} value Any entity
+ * @return {boolean} - true if yes, false otherwise.
+ */
+export function isDate(value: any) {
+  return Object.prototype.toString.call(value) == "[object Date]";
+}
+
+/**
+ * Determines whether the given `value` is a `string` object or not.
+ * @param {any} value Any entity
+ * @return {boolean} - true if yes, false otherwise.
+ */
+export function isString(value: any) {
+  return Object.prototype.toString.call(value) == "[object String]";
+}
+
+/**
+ * Determines whether the given `value` is a `Object` or not.
+ * @param {any} value Any entity
+ * @return {boolean} - true if yes, false otherwise.
+ */
+export const isObject = function(value: any) {
+  return value === Object(value);
+};
+
+/**
+ * Computes byte length of given string assuming the encoding to use is "utf-8"
+ * @param str
+ */
+export function byteLength(str: string) {
```

Comment:
This was copied over verbatim because rollup config wasn't playing nice with importing the browser implementation. Agree that usage can be simplified with removing this, since only usage is checking if response.bodyAsText.length > 0 or not.